src/transport/mod.rs

```diff
@@ -82,6 +83,12 @@ pub(crate) trait Publish {
     async fn unset(&mut self, data: ValidatedUnset) -> Result<(), crate::Error>;
 }
 
+pub(crate) type EventResult<P> = Result<Option<Result<ReceivedEvent<P>, RecvError>>, crate::Error>;
+pub(crate) type DeserializeIndResult =
+    Result<Result<Option<(AstarteType, Option<Timestamp>)>, RecvError>, crate::Error>;
+pub(crate) type DeserializeObjResult =
```

Comment:
This is really confusing. We are having multiple layers of errors and values.

This signature can be flattened by grouping the errors. The result should be:

```rust
pub(crate) enum TransportError {
    /// Error that will be sent to the client
    Recv(#[from] RecvError),
    /// Error from the underline transport
    Transport(#[source] crate::Error),
}

fn foo() -> Result<Option<T>, ConnectionError> {...}
```